internal/dump/json.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package dump
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type DumpableJSONResource interface {
+	Name() string
+	JSON() []byte
+}
+
+func dumpJSONResource(dir string, object DumpableJSONResource) error {
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("failed to create dump directory: %w", err)
+	}
+	formatted, err := formatJSON(object.JSON())
```

Comment:
There is `formatter` package in elastic-package, maybe you can use it here.

Reply:
Oh, ok! Better reuse the formatter package.
I'll use it and update all the dumped files.
Thanks!

Reply:
According to this https://github.com/elastic/elastic-package/pull/862#discussion_r908489425 , I'll keep using this method (formatJSON) to write the files.

Reply:
Yes, please do :)